addons/web/static/src/core/install_prompt/install_prompt_service.js

```diff
@@ -0,0 +1,79 @@
+/** @odoo-module **/
+
+import { reactive } from "@odoo/owl";
+import { browser } from "@web/core/browser/browser";
+import { isDisplayStandalone, isIOS, isMacOS } from "@web/core/browser/feature_detection";
+import { registry } from "@web/core/registry";
+import { InstallPrompt } from "./install_prompt";
+
+const serviceRegistry = registry.category("services");
+
+const installPromptService = {
+    dependencies: ["dialog"],
+    start(env, { dialog }) {
+        let nativePrompt;
+
+        const state = reactive({
+            canPromptToInstall: false,
+            decline,
+            show,
+        });
+
+        // The PWA can only be installed if the app is not already launched
+        // The installation is not possible from the tests by default
+        // Finally, PWA are supported on desktop Safari since version 17
+        const canBeInstalled =
+            !isDisplayStandalone() &&
+            browser.location.pathname !== "/web/tests" &&
+            (browser.BeforeInstallPromptEvent !== undefined ||
+                isIOS() ||
+                (isMacOS() && browser.navigator.userAgent.match(/Version\/(\d+)/)[1] >= 17));
```

Comment:
I don't really know how we could detect that we are running on a version of Safari that has the 'Pin to dock' option available in the application menu, since there is no API related to that specifically

Reply:
To be more precise, it is only available on Safari 17 + on macOS Sonoma

Reply:
I see your point.
On a side note, wouldn't `iOS() || isMacOS()` also match other browsers than Safari on these platforms ? :thinking: 
Should it also apply on those non-Safari browsers ? And if not, maybe `isBrowserSafari()` is more adequate ?

Reply:
Are you sure about your logic for desktop Safari 17+ ?